### PR TITLE
Table for scalar variables

### LINKED ISSUE #1694: Bad error in sc.table with 0D variable
```.py
sc.table(sc.Variable(value=0))
```
produces
```.py
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-9-4f0f55b7fd6a> in <module>
----> 1 sc.table(sc.Variable(value=0))

~/Work/scipp-devel/install/scipp/table.py in table(scipp_obj)
    201 
    202     from IPython.display import display
--> 203     tv = TableViewer(scipp_obj)
    204     display(tv.box)
    205 

~/Work/scipp-devel/install/scipp/table.py in __init__(self, scipp_obj)
    260                 if su.is_variable(scipp_obj):
    261                     self.headers = 1
--> 262                     key = str(scipp_obj.dims[0])
    263                     var = scipp_obj
    264                 else:

IndexError: list index out of range
```

Should `table` be able to show 0D Variables? Or should the error me more descriptive?

It feels like we had this one before?

### PR DESCRIPTION
**Pull request by OwenArnold**
fixes #1694

Enables scalar variables to be shown as single rows with no dimension information

### CODE REVIEW

File: python/tests/test_table.py [hunk 1]
```diff
@@ -182,3 +182,7 @@ def test_display_when_largest_coord_non_dimensional():
                                        values=np.random.random(10),
                                        unit=sc.units.counts))
     sc.table(da)
+
+
+def test_with_scalar():
+    sc.table(sc.scalar(value=1.0))
```
Comment (nvaytet) [inline python/tests/test_table.py:188]: what happens if there are multiple scala variables, e.g. in a Dataset?

**Review by nvaytet — COMMENTED** [review]
Maybe add the bug fix to the release notes?

**Comment by SimonHeybrock** [comment]
> Maybe add the bug fix to the release notes?

Maybe something we should discuss, but personally I'd prefer listing only really relevant changes in the release notes, to keep them concise. The longer they get the less likely someone will actually read them.

File: python/tests/test_table.py (hunk 1 above)
Comment (OwenArnold) [reply python/tests/test_table.py:188]: Good question. I didn't try that. I did with DataArray, but i'll add tests and check outputs for all.

**Comment by OwenArnold** [comment]
Outputs for scalars in various scipp types look like this:
image

**Pull request merged**